### ISSUE TITLE
Vanilla or React.js

### ISSUE THREAD

**Issue by N3v1** [TeamCodeArena/deliverease.app#3]
Hi @SID262000, @Devansh-Singh-Parmar, everyone,
Should we use HTML, CSS, and JS for the website or React.js (TypeScriptReact)? I've heard that React is better and easier for a well-designed website. Please let me/us know what you think about what we should use

Best regards
NH

**Comment by SID262000** [comment]
My perspective says that we should always stick to the foundational building blocks, HTML, CSS, JS to keep the codebase really robust and scalable.
cc: @N3v1

**Comment by N3v1** [comment]
> My perspective says that we should always stick to the foundational building blocks, HTML, CSS, JS to keep the codebase really robust and scalable. cc: @N3v1

Thank you for your point of view @SID262000

**Comment by Devansh-Singh-Parmar** [comment]
I agree with @SID262000 , But incase of changing this react will be my first priority.
cc @N3v1 @SID262000

**Comment by jiri132** [comment]
Hey everyone!!

To make a well designed website with the standard building blocks is harder, but scalable.

I have worked with React Native in the past, it looks daunting to start using it!, BUT there is a huge trade off with impressive fast scalability if you design the code correctly.

To keep it simple:
1. Scalability + well-written code = React
2. Available for everyone = html css (js / ts)

**Comment by KartikKar19** [comment]
@N3v1 I think that React would be a better alternative as we have less time and more work to do but the issue is I don't know React. If somebody knows React then it would be easier for them.

**Comment by N3v1** [comment]
I have some basic knowledge of React, but limited experience with it. I could try to code with TypeScriptReact (tsx).

I'd like to express my gratitude to you, @jiri132, for assisting us. Thank you so much! 🙏
cc: @jiri132, @KartikKar19 , @SID262000

**Comment by N3v1** [comment]
> Hey everyone!!
> 
> To make a well designed website with the standard building blocks is harder, but scalable.
> 
> I have worked with React Native in the past, it looks daunting to start using it!, BUT there is a huge trade off with impressive fast scalability if you design the code correctly.
> 
> To keep it simple:
> 
> 1. Scalability + well-written code = React
> 2. Available for everyone = html css (js / ts)

@jiri132 After we finish designing the page, will you be available for its development? Maybe I could help you out.

**Comment by SID262000** [comment]
Hi @N3v1 ,
We really need to brainstorm this alongside @jiri132 to arrive at a constructive solution at the end of the day.
Welcome onboard @jiri132, glad to have you on the Team.
I'm not sure if you have undergone the onboarding process, please confirm accordingly.

**Comment by SID262000** [comment]
> Hey everyone!!
> 
> To make a well designed website with the standard building blocks is harder, but scalable.
> 
> I have worked with React Native in the past, it looks daunting to start using it!, BUT there is a huge trade off with impressive fast scalability if you design the code correctly.
> 
> To keep it simple:
> 1. Scalability + well-written code = React
> 2. Available for everyone = html css (js / ts)
> 

All the frameworks including React, Angular, Node are built on top of the foundational building blocks of HTML CSS and JS. I totally agree with the fact of the hardwork required in building a website using the foundational building blocks, but my experience says that if we really want it to be scalable, we really need to stick to the foundational building blocks!
cc: @jiri132 @N3v1

**Comment by N3v1** [comment]
Please vote here https://github.com/TeamCodeArena/deliverease.app/discussions/6 

cc: @KartikKar19 @Devansh-Singh-Parmar @jiri132 @SID262000

**Comment by N3v1** [comment]
The vote #6 did not give a reasonable result, I would say that we first program with HTML, CSS, JS/TS and if we still need React, we will add that. What do you think of it?

cc: @SID262000 @Devansh-Singh-Parmar @jiri132 @Atrika10 @AryanGupta0007

**Comment by SID262000** [comment]
I completely agree with @N3v1 
Clear GREEN SIGNAL from my end, go for it!

**Comment by AryanGupta0007** [comment]
@N3v1 changing to react means rewriting the whole backend so please keep that in mind. @SID262000

**Comment by N3v1** [comment]
@AryanGupta0007 What do you think? Should we use React it not?

**Comment by AryanGupta0007** [comment]
@N3v1 using react is great but as i mentioned it will need a lot of changes in backend. 
Cc: @SID262000

**Comment by SID262000** [comment]
Requesting everyone to not divert themselves into React or other such frameworks at the moment, that will require a lot of effort.
As clearly indicated in the preceding thread, stick to HTML CSS and JS/TS for building.

Thanks everyone for your active contribution!

**Comment by N3v1** [comment]
Okay @SID262000

**Comment by Devansh-Singh-Parmar** [comment]
I agree with @SID262000 & @AryanGupta0007 for now, we can look into this matter once we complete the development phase and all.
cc @N3v1
